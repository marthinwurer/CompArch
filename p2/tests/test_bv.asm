;
; tests BVC/BVS
;
; uses instructions MOV, BPT, ADD, HALT
;
; uses addressing modes 0, 2, 6
;
	.globl	start
;
v1:	.word	0
v2:	.word	100000
v3:	.word	077777
;
start:
	mov	v3,r3
	add	#1,r3
	bvs	ok1
	add	#1,r0
ok1:
	add	#1,r3
	bvc	ok2
	add	#2,r0
ok2:
	mov	v2,r2
	add	#1,r2
	bvs	bad3
	jmp	ok3
bad3:
	add	#4,r0
ok3:
	add	#1,r2
	bvc	ok4
	add	#10,r0
ok4:
;
	bpt
	halt
;
	.end	start
