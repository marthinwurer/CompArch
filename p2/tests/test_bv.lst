      1                                 ;
      2                                 ; tests BVC/BVS
      3                                 ;
      4                                 ; uses instructions MOV, BPT, ADD, HALT
      5                                 ;
      6                                 ; uses addressing modes 0, 2, 6
      7                                 ;
      8 000000                                   .globl  start
      9                                 ;
     10 000000  000000                  v1:      .word   0
     11 000002  100000                  v2:      .word   100000
     12 000004  077777                  v3:      .word   077777
     13                                 ;
     14 000006                          start:
     15 000006  016703  177772                   mov     v3,r3
     16 000012  062703  000001                   add     #1,r3
     17 000016  102402                           bvs     ok1
     18 000020  062700  000001                   add     #1,r0
     19 000024                          ok1:
     20 000024  062703  000001                   add     #1,r3
     21 000030  102002                           bvc     ok2
     22 000032  062700  000002                   add     #2,r0
     23 000036                          ok2:
     24 000036  016702  177740                   mov     v2,r2
     25 000042  062702  000001                   add     #1,r2
     26 000046  102402                           bvs     bad3
     27 000050  000167  000004                   jmp     ok3
     28 000054                          bad3:
     29 000054  062700  000004                   add     #4,r0
     30 000060                          ok3:
     31 000060  062702  000001                   add     #1,r2
     32 000064  102002                           bvc     ok4
     33 000066  062700  000010                   add     #10,r0
     34 000072                          ok4:
     35                                 ;
     36 000072  000003                           bpt
     37 000074  000000                           halt
     38                                 ;
     39 000076  000006'                          .end    start


bad3     000054 R  
ok1      000024 R  
ok2      000036 R  
ok3      000060 R  
ok4      000072 R  
pc     = 000007 %  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000006 RG 
v1       000000 R  
v2       000002 R  
v3       000004 R  
