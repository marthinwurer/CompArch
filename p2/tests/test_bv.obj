000 2 00 00
002 2 00 80
004 2 ff 7f
006 4 c3 1d fa ff
00a 4 c3 65 01 00
00e 2 02 85
010 4 c0 65 01 00
014 4 c3 65 01 00
018 2 02 84
01a 4 c0 65 02 00
01e 4 c2 1d e0 ff
022 4 c2 65 01 00
026 2 02 85
028 4 77 00 04 00
02c 4 c0 65 04 00
030 4 c2 65 01 00
034 2 02 84
036 4 c0 65 08 00
03a 2 03 00
03c 2 00 00
006
