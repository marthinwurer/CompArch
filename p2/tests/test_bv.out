CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 6
000006:  PS[00]  016703 177772  MOV 177772(R7),R3
000012:  PS[00]  062703  ADD (R7)+,R3
000016:  PS[12]  102402  BVS 4
000024:  PS[12]  062703  ADD (R7)+,R3
000030:  PS[10]  102002  BVC 4
000036:  PS[10]  016702 177740  MOV 177740(R7),R2
000042:  PS[10]  062702  ADD (R7)+,R2
000046:  PS[10]  102402  BVS 4
000050:  PS[10]  000167 000004  JMP 000004(R7)
000060:  PS[10]  062702  ADD (R7)+,R2
000064:  PS[10]  102002  BVC 4
000072:  PS[10]  000003  BPT
  R0[000000]  R1[000000]  R2[100002]  R3[100001]
  R4[000000]  R5[000000]  R6[000000]  R7[000074]
000074:  PS[10]  000000  HALT

Machine Halted - HALT instruction

Simulated time 99 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
